src/main/collectors/AbstractCollector.ts

```diff
@@ -35,6 +35,24 @@ export class AbstractCollector {
 		// override in subclass
 	}
 
+	/**
+	 * Used to log if a handler fails its execution
+	 * Usage: document.addEventListener("click", this.logWrapHandler(yourhandler, logger))
+	 * @param handler
+	 * @param log
+	 * @param handlerArgs
+	 * @protected
+	 */
+	protected logWrapHandler(handler: Function, log: Logger, ...handlerArgs) {
```

Comment:
Is this for covering all cases, including where the the handler is not a descendant of AbstractCollector? Wouldn't it be better to encapsulate the logic in AbstractCollector, this would avoid having to place the Wrap call everywhere.

Reply:
what exactly do you think of? somethink like a `addEventHandler(eventName, callback, logger, ...args)`?
In the end it would be the same thing :thinking: